packages/playwright/src/runner/watchMode.ts

```diff
@@ -316,15 +338,9 @@ Change settings
       case 'f': result.resolve('failed'); break;
       case 's': result.resolve('toggle-show-browser'); break;
     }
-  };
-
-  process.stdin.on('keypress', handler);
-  void result.finally(() => {
-    process.stdin.off('keypress', handler);
-    rl.close();
-    if (process.stdin.isTTY)
-      process.stdin.setRawMode(false);
   });
+
+  void result.finally(reader.dispose);
```

Comment:
I think readKeyPress should have this code, not readCommand

Reply:
i've moved `readKeyPress` to return a promise in https://github.com/microsoft/playwright/pull/32584/commits/f0de6ccd81ff07f9e947625839b3335e0da3ba4a. That changes quite a bit about the code, but makes it easier to reason about in general - and it also moves the disposal into `readKeyPress` 👍